### DISCUSSION TITLE
Applying, Deploying a specific source configuration type to one or more environments

### DISCUSSION THREAD

**Discussion by leonvzGit** [Dynatrace/dynatrace-configuration-as-code#215] [Archive]
**Is your feature request related to a problem? Please describe.**
Given a projects folder I created from downloading the configuration from 3 different environments in the environments.yaml file:
root/
  CentralDEV/
    alerting-profile/
    management-zone/
    dashboard/     
  CentralTEST/
    alerting-profile/
    ... 
  CentralQA/
    ...    
environments.yaml
 
Given the following command:
monaco deploy --environments=environments.yaml -specific-environment="CentralDEV" -p="CentralTEST/alerting-profile" --dry-run
 
There are no configurations being picked up to apply on CentralDEV
 
Does this mean if I want to export a specific configuration type(in this case alerting profile) from a single environment to another, I would have to create a project directory containing only the configuration type folder eg alerting profile?

**Describe the solution you'd like**
I would like the -p flag to accept the configuration type folder eg project/configuration-type, or a new flag to be introduced for a configuration-type that you want to apply. So that I can use exported configuration and sync specific configuration types from downloaded environments. This ties into the download command having the flag --downloadSpecificAPI. I would like to use a similar flag when deploying.

**Describe alternatives you've considered**

**Additional context**
Add any other context about the feature request here.

**Comment by didiladi** [comment]
@leonvzGit yes, you are right. In this case you would have to create a separate project with just the alerting profile in it.

The folder "alerting-profile" is treated as a set of configurations - its not a project.

@dynatrace-oss/mac-maintainers might be a good idea to offer a param to deploy just a single configuration-type (e.g. alerting-profiles).

**Reply by tatomir146** [reply]
Monaco should be used to reflect configuration stored in repository to the cluster, not only a part of it. For now, we don't have plans to support deploying only one specific type of config.

**Reply by mdicss** [reply]
deploy should have a parameter --deploySpecificAPI analogous to --downloadSpecificAPI in the download command. So it would be possible to download a changed configuration and deploy it to another environment more easyly.

**Comment by interceptor** [comment]
I have spent 2 days trying to get this to work also:
"monaco deploy --environments=environments.yaml -specific-environment="CentralDEV" -p="CentralTEST/alerting-profile" --dry-run"

The documentation is very poor and misleading.

"The folder "alerting-profile" is treated as a set of configurations - its not a project."

What on earth is a project then?? Whatever is going on HERE: https://dynatrace-oss.github.io/dynatrace-monitoring-as-code/configuration/deploy_configuration needs some serious clarifications.

**Reply by UnseenWizzard** [reply]
Hi @interceptor - I agree that is sadly not cleary defined in the current documentation. 

In general there is a distinction between a "_project_" folder and a "_configuration_" folder. 

A _configuration_ folder is one named after a specific [type](https://dynatrace-oss.github.io/dynatrace-monitoring-as-code/configuration/configTypes_tokenPermissions) - like `alerting-profile`- and generally contains a yaml configuration and one ore more json templates.

A _project_ folder is one in which you group several configuration folders together. 

I hope that clarifies what was meant by "The folder "alerting-profile" is treated as a set of configurations - its not a project."

If you take a look at this test config you will see three projects: 
https://github.com/dynatrace-oss/dynatrace-monitoring-as-code/tree/main/cmd/monaco/test-resources/integration-multi-project

I don't know how the rest of your configuration/folder structure, but from the command snippet you've posted "CentralTEST" would be the "project", and "alerting-profile" one set of alerting profile configurations within in. 

Hope that helps? 
If the wording of my description makes enough sense to you, I'll add that to the documentation - else please let me know what would help to further clarify this
